modules/misc/test/tests/ad_arrhenius_material_property/tests

```diff
@@ -0,0 +1,23 @@
+[Tests]
+  [exact]
+    type = 'CSVDiff'
+    input = 'exact.i'
+    csvdiff = 'exact_out.csv'
+    issues = '#22384'
+    design = 'ArrheniusMaterialProperty.md'
+    requirement = 'The system shall compute a generic material property based on an Arrhenius model that matches an analytic solution of that equation while using the automatic differentiation system'
+  []
+  [exact-jac]
+    type = 'PetscJacobianTester'
+    input = 'exact.i'
+    run_sim = 'True'
+    cli_args = "Outputs/active=''"
+    prereq = exact
+    petsc_version = '>=3.9'
+    ratio_tol = 3e-7
+    difference_tol = 1
+    issues = '#22384'
+    design = 'jacobian_definition.md'
+    requirement = 'The system shall provide perfect derivatives using ADArrheniusMaterialProperty.'
+  []
+[]
```

Comment:
```suggestion
[Tests]
  issues = '#22384'
  design = 'ArrheniusMaterialProperty.md'
  [exact]
    type = 'CSVDiff'
    input = 'exact.i'
    csvdiff = 'exact_out.csv'
    requirement = 'The system shall compute a generic material property based on an Arrhenius model that matches an analytic solution of that equation while using the automatic differentiation system.'
  []
  [exact-jac]
    type = 'PetscJacobianTester'
    input = 'exact.i'
    run_sim = 'True'
    cli_args = "Outputs/active=''"
    prereq = exact
    petsc_version = '>=3.9'
    ratio_tol = 3e-7
    difference_tol = 1
    requirement = 'The system shall provide perfect derivatives when using ADArrheniusMaterialProperty.'
  []
[]
```

Let's not use `jacobian_definition.md` here, since that will require lots of follow-on changes to quite a number of apps. Instead, let's use the object documentation (and issue number) at the `[Tests]` level.